Add test parameter to performance dashboard to support more customization in A/B testing.

dewei-zhu opened the pull request:
#### 473724c9a49c6b95dc3104784c2eb7fbc0c0764b
<pre>
Add test parameter to performance dashboard to support more customization in A/B testing.
<a href="https://bugs.webkit.org/show_bug.cgi?id=257017">https://bugs.webkit.org/show_bug.cgi?id=257017</a>
rdar://100645382

Reviewed by Alexey Proskuryakov.

Introduce &apos;test parameter&apos; concept to performance dashboard which adds more customization ability.
Create 1-to-N relationship between triggerable configuration and test parameter so that different
(platform, test) pair can support different test parameters.
Add test parameter set which stores a set of parameter value and build requests can optionally refer to it.

* Websites/perf.webkit.org/init-database.sql: Added &apos;test_parameters&apos; and its related tables.
Added &apos;test_parameter_sets&apos; and a corresponding field in &apos;build_requests&apos; table that refers to it.
* Websites/perf.webkit.org/migrate-database.sql: Added migration SQL queries.
* Websites/perf.webkit.org/public/admin/test-parameters.php: Added an admin page for &apos;test_parameters&apos; table.
* Websites/perf.webkit.org/public/api/build-requests.php: Added &apos;testParameterSets&apos; to API return value.
* Websites/perf.webkit.org/public/api/test-groups.php: Added &apos;testParameterSets&apos; to API return value.
* Websites/perf.webkit.org/public/api/update-triggerable.php: Added support for updating test parameters
for triggerable configurations.
* Websites/perf.webkit.org/public/include/admin-header.php: Added &apos;Test Parameters&apos; tab.
Updated &apos;assert&apos; usage to avoid using deprecated syntax.
Fixed a bug in &apos;update_boolean_field&apos; that updating a boolean column on a table with multile boolean
columns only works for first column.
Added &apos;select&apos; type in &apos;render_form_control_for_column&apos;.
* Websites/perf.webkit.org/public/include/build-requests-fetcher.php: Added &apos;testParameterSets&apos; to
API return value.
* Websites/perf.webkit.org/public/include/commit-sets-helpers.php: Added code to support inserting
test parameters to build requests while creating a test group.
* Websites/perf.webkit.org/public/include/db.php: Added helper function to be able to fetch enum values.
* Websites/perf.webkit.org/public/include/manifest-generator.php: Added &apos;testParameters&apos; to manifest.
Added test parameter and triggerable configuration relationship in manifest.
* Websites/perf.webkit.org/public/privileged-api/add-build-requests.php: Build requests added using this API
should created with the same test parameter sets as the original build requests.
* Websites/perf.webkit.org/public/privileged-api/create-analysis-task.php: Added &apos;testParametersList&apos; arugment
to this API and corresponding test parameter sets will be created.
* Websites/perf.webkit.org/public/privileged-api/create-test-group.php: Added &apos;testParametersList&apos; argument
to this API and correspoding test parameter sets will be created.
* Websites/perf.webkit.org/public/v3/components/custom-configuration-test-group-form.js: Adopted new API.
(CustomConfigurationTestGroupForm.prototype.startTesting):
* Websites/perf.webkit.org/public/v3/index.html: Added &apos;test-parameter.js&apos;.
* Websites/perf.webkit.org/public/v3/models/build-request.js: Added &apos;testParameterSet&apos;.
(BuildRequest):
(BuildRequest.prototype.updateSingleton):
(BuildRequest.prototype.testParameterSet):
(BuildRequest.prototype.async findBuildRequestWithSameRoots):
(BuildRequest.constructBuildRequestsFromData):
* Websites/perf.webkit.org/public/v3/models/commit-set.js: Removed an extra space.
(CommitSet.prototype.commits):
* Websites/perf.webkit.org/public/v3/models/manifest.js: Added code to resolve test parameters.
(Manifest.reset):
(Manifest._didFetchManifest):
(Manifest):
* Websites/perf.webkit.org/public/v3/models/measurement-adaptor.js: Added testParameterSet function which currently
returns null.
* Websites/perf.webkit.org/public/v3/models/test-group.js: Extend API to support optional test parameter sets.
(TestGroup):
(TestGroup.prototype.addBuildRequest):
(TestGroup._computeRequestedCommitSets.set return):
(TestGroup._computeRequestedCommitSets.set requestedCommitSets):
(TestGroup._computeRequestedCommitSets.const.set request):
(TestGroup._computeRequestedCommitSets):
(TestGroup.set const):
(TestGroup.async createWithTask):
(TestGroup.async createWithCustomConfiguration):
(TestGroup.async createAndRefetchTestGroups):
* Websites/perf.webkit.org/public/v3/models/test-parameter.js: Added &apos;TestParameter&apos; and &apos;TestParameterSet&apos;
models which are JavaScript representations of &apos;test_paramters&apos; and &apos;test_parameter_sets&apos; tables.
Added &apos;CustomTestParameterSet&apos; to be used in the UI.
(TestParameter):
(TestParameter.prototype.get disabled):
(TestParameter.prototype.get type):
(TestParameter.prototype.get description):
(TestParameter.prototype.get hasFile):
(TestParameter.prototype.get hasValue):
(TestParameter.findByName):
(TestParameterSet):
(TestParameterSet.prototype.updateSingleton):
(TestParameterSet.prototype._updateFromObject):
(TestParameterSet.prototype.get buildTypeParameters):
(TestParameterSet.prototype.get testTypeParameters):
(TestParameterSet.prototype.get parameters):
(TestParameterSet.prototype.asPayload):
(TestParameterSet.prototype.hasSameBuildParametersAs):
(TestParameterSet.prototype.equals):
(TestParameterSet.prototype.valueForParameter):
(TestParameterSet.prototype.fileForParameter):
(TestParameterSet.prototype.valueForParameterName):
(TestParameterSet.prototype.fileForParameterName):
(TestParameterSet.prototype._createPayloadForParameters):
(TestParameterSet.prototype._checkEquals):
(TestParameterSet.isValueEqual):
(TestParameterSet.equals):
(TestParameterSet.hasSameBuildParameters):
(CustomTestParameterSet):
(CustomTestParameterSet.prototype.get buildTypeParameters):
(CustomTestParameterSet.prototype.get testTypeParameters):
(CustomTestParameterSet.prototype.get parameters):
(CustomTestParameterSet.prototype.asPayload):
(CustomTestParameterSet.prototype.valueForParameter):
(CustomTestParameterSet.prototype.fileForParameter):
(CustomTestParameterSet.prototype.valueForParameterName):
(CustomTestParameterSet.prototype.fileForParameterName):
(CustomTestParameterSet.prototype.set):
(CustomTestParameterSet.prototype.get):
(CustomTestParameterSet.prototype.delete):
(CustomTestParameterSet.prototype.setByName):
(CustomTestParameterSet.prototype.getByName):
(CustomTestParameterSet.prototype._createPayloadForParameters):
* Websites/perf.webkit.org/public/v3/models/triggerable.js: Added &apos;testParamters&apos; to &apos;TriggerableConfiguration&apos; to reflect
corresponding database change.
(Triggerable):
(prototype.get testParameters):
(createForTriggerable):
* Websites/perf.webkit.org/public/v3/pages/analysis-task-page.js: Adopted APIs that create test groups now require test parameter sets to be specified.
(AnalysisTaskPage.prototype.async _retryCurrentTestGroup):
(AnalysisTaskPage.prototype.async _bisectCurrentTestGroup):
(AnalysisTaskPage.set const):
(AnalysisTaskPage.prototype.async _createTestGroupAfterVerifyingCommitSetList):
(AnalysisTaskPage.prototype.async _createCustomTestGroup):
* Websites/perf.webkit.org/public/v3/pages/create-analysis-task-page.js:
(CreateAnalysisTaskPage.prototype.async _createAnalysisTaskWithGroup):
* Websites/perf.webkit.org/server-tests/api-build-requests-tests.js:
(then):
(async await):
* Websites/perf.webkit.org/server-tests/api-manifest-tests.js:
(return.TestServer.remoteAPI.getJSON.string_appeared_here.then):
* Websites/perf.webkit.org/server-tests/api-test-groups.js:
(async const):
* Websites/perf.webkit.org/server-tests/api-update-triggerable-tests.js: Updated and added unit tests.
* Websites/perf.webkit.org/server-tests/privileged-api-add-build-requests-tests.js: Updated and added unit tests.
(async addTriggerableAndCreateTask):
(async await):
(webkit.id):
(const.commitSet.of.updatedGroups.0.requestedCommitSets):
* Websites/perf.webkit.org/server-tests/privileged-api-create-analysis-task-tests.js: Updated and added unit tests.
* Websites/perf.webkit.org/server-tests/privileged-api-create-test-group-tests.js: Updated and added unit tests.
(async await):
(webkit.id):
(async const):
* Websites/perf.webkit.org/server-tests/resources/mock-data.js: Added test parameters related mock data.
(MockData.addMockConfiguration):
(MockData.async await):
(MockData.set addMockDataWithBuildAndTestTypeTestParameterSets):
(MockData.set addMockDataWithTestParameterSets):
(MockData.mockTestSyncConfigWithTestParameters):
(MockData.addEmptyTriggerable): Deleted.
(MockData.addMockTestGroupWithGitWebKit): Deleted.
* Websites/perf.webkit.org/server-tests/tools-buildbot-triggerable-tests.js: Updated and added unit tests.
(async const):
* Websites/perf.webkit.org/server-tests/tools-sync-buildbot-integration-tests.js:
(configWithOneTesterTwoBuilders):
* Websites/perf.webkit.org/tools/js/buildbot-syncer.js:
(BuildbotSyncer):
(BuildbotSyncer.prototype.addTestConfiguration):
(BuildbotSyncer.prototype.addBuildConfiguration):
(BuildbotSyncer.prototype.matchesConfiguration):
(BuildbotSyncer.prototype._propertiesForBuildRequest):
(BuildbotSyncer._loadConfig):
(BuildbotSyncer._resolveBuildersWithPlatforms):
(BuildbotSyncer._validateAndMergeConfig):
* Websites/perf.webkit.org/tools/js/buildbot-triggerable.js: Updated and added unit tests.
(BuildbotTriggerable.prototype.updateTriggerable):
* Websites/perf.webkit.org/tools/js/database.js:
* Websites/perf.webkit.org/tools/js/v3-models.js:
* Websites/perf.webkit.org/unit-tests/build-request-tests.js: Updated and added unit tests.
(sampleBuildRequestData):
(oneTestGroup):
(threeTestGroups):
* Websites/perf.webkit.org/unit-tests/resources/mock-v3-models.js: Added mock test parameters.
* Websites/perf.webkit.org/unit-tests/test-groups-tests.js: Updated and added unit tests.
(sampleTestGroup):
(sampleTestGroupForRetry):
* Websites/perf.webkit.org/unit-tests/test-parameter-tests.js: Added unit tests.

Canonical link: <a href="https://commits.webkit.org/265477@main">https://commits.webkit.org/265477@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a860350007bc3cb28e608ade2751ecd7861d46b4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10989 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/11202 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/11521 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/12636 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/10494 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/11005 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/13579 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/11184 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/13422 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/11148 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/12046 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/9262 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/13042 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/9339 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/9932 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/17151 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/9304 "Build is in progress. Recent messages:") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/10410 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/10086 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/13316 "Passed tests") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/10522 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/8611 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9694 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2639 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13963 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/10376 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

xw commented on Websites/perf.webkit.org/public/v3/models/triggerable.js:
```diff
@@ -52,18 +53,20 @@ class TriggerableConfiguration extends DataModelObject {
         this.#platform = object.platform;
         console.assert(object.supportedRepetitionTypes.every(TriggerableConfiguration.isValidRepetitionType.bind(TriggerableConfiguration)));
         this.#supportedRepetitionTypes = object.supportedRepetitionTypes;
+        this.#testParameters = object.testParameters;
```
Nit: Are we guaranteed to always have `object.testParameters` be defined? If not, it may be worth having a fallback of an empty dict here (or in the getter method)

dewei-zhu replied:
Yes, check `v3/models/manifest.js` around line 95.

xw commented on Websites/perf.webkit.org/public/v3/models/test-group.js:
```diff
@@ -160,6 +158,16 @@ class TestGroup extends LabeledObject {
+
+    requestedTestParameterSets()
+    {
+        return this.requestedCommitSets().map(commitSet => this.requestsForCommitSet(commitSet)[0].testParameterSet());
```
Nit: This snippet does not chain optionals and the one in `testParameterSetForCommitSet()` above does - should this also be using optionals in case `requestsForCommitSet` can't find the commitSet requested? (or alternatively - can we reuse `testParameterSetForCommitSet` here?)

xw commented on Websites/perf.webkit.org/public/v3/models/test-parameter.js:
```diff
@@ -0,0 +1,212 @@
+    }
+}
+
+class CustomTestParameterSet {
```
Little confused as to what the difference between `TestParameterSet` and `CustomTestParameterSet` is - is it that the former is intended to be used as predefined parameters in the backend, and the latter for the frontend? If this is the case, should `CustomTestParameterSet` inherit `TestParameterSet`?

dewei-zhu replied:
Yes, `CustomTestParameterSet` is for front end editing to craft payload sent to backend API.
`CustomTestParameterSet` cannot inherit `TestParameterSet` because `TestParameterSet` inherits `DataModelObject` which presents database entries (has `id` field).

xw reviewed (approved):
Unofficial r=me

aproskuryakov reviewed (approved):
rs=me. This may be the biggest amount of plumbing code I've seen that was needed to pass an argument!

dewei-zhu committed:
Add test parameter to performance dashboard to support more customization in A/B testing.
https://bugs.webkit.org/show_bug.cgi?id=257017
rdar://100645382

Reviewed by Alexey Proskuryakov.

Introduce 'test parameter' concept to performance dashboard which adds more customization ability.
Create 1-to-N relationship between triggerable configuration and test parameter so that different
(platform, test) pair can support different test parameters.
Add test parameter set which stores a set of parameter value and build requests can optionally refer to it.

* Websites/perf.webkit.org/init-database.sql: Added 'test_parameters' and its related tables.
Added 'test_parameter_sets' and a corresponding field in 'build_requests' table that refers to it.
* Websites/perf.webkit.org/migrate-database.sql: Added migration SQL queries.
* Websites/perf.webkit.org/public/admin/test-parameters.php: Added an admin page for 'test_parameters' table.
* Websites/perf.webkit.org/public/api/build-requests.php: Added 'testParameterSets' to API return value.
* Websites/perf.webkit.org/public/api/test-groups.php: Added 'testParameterSets' to API return value.
* Websites/perf.webkit.org/public/api/update-triggerable.php: Added support for updating test parameters
for triggerable configurations.
* Websites/perf.webkit.org/public/include/admin-header.php: Added 'Test Parameters' tab.
Updated 'assert' usage to avoid using deprecated syntax.
Fixed a bug in 'update_boolean_field' that updating a boolean column on a table with multile boolean
columns only works for first column.
Added 'select' type in 'render_form_control_for_column'.
* Websites/perf.webkit.org/public/include/build-requests-fetcher.php: Added 'testParameterSets' to
API return value.
* Websites/perf.webkit.org/public/include/commit-sets-helpers.php: Added code to support inserting
test parameters to build requests while creating a test group.
* Websites/perf.webkit.org/public/include/db.php: Added helper function to be able to fetch enum values.
* Websites/perf.webkit.org/public/include/manifest-generator.php: Added 'testParameters' to manifest.
Added test parameter and triggerable configuration relationship in manifest.
* Websites/perf.webkit.org/public/privileged-api/add-build-requests.php: Build requests added using this API
should created with the same test parameter sets as the original build requests.
* Websites/perf.webkit.org/public/privileged-api/create-analysis-task.php: Added 'testParametersList' arugment
to this API and corresponding test parameter sets will be created.
* Websites/perf.webkit.org/public/privileged-api/create-test-group.php: Added 'testParametersList' argument
to this API and correspoding test parameter sets will be created.
* Websites/perf.webkit.org/public/v3/components/custom-configuration-test-group-form.js: Adopted new API.
(CustomConfigurationTestGroupForm.prototype.startTesting):
* Websites/perf.webkit.org/public/v3/index.html: Added 'test-parameter.js'.
* Websites/perf.webkit.org/public/v3/models/build-request.js: Added 'testParameterSet'.
(BuildRequest):
(BuildRequest.prototype.updateSingleton):
(BuildRequest.prototype.testParameterSet):
(BuildRequest.prototype.async findBuildRequestWithSameRoots):
(BuildRequest.constructBuildRequestsFromData):
* Websites/perf.webkit.org/public/v3/models/commit-set.js: Removed an extra space.
(CommitSet.prototype.commits):
* Websites/perf.webkit.org/public/v3/models/manifest.js: Added code to resolve test parameters.
(Manifest.reset):
(Manifest._didFetchManifest):
(Manifest):
* Websites/perf.webkit.org/public/v3/models/measurement-adaptor.js: Added testParameterSet function which currently
returns null.
* Websites/perf.webkit.org/public/v3/models/test-group.js: Extend API to support optional test parameter sets.
(TestGroup):
(TestGroup.prototype.addBuildRequest):
(TestGroup._computeRequestedCommitSets.set return):
(TestGroup._computeRequestedCommitSets.set requestedCommitSets):
(TestGroup._computeRequestedCommitSets.const.set request):
(TestGroup._computeRequestedCommitSets):
(TestGroup.set const):
(TestGroup.async createWithTask):
(TestGroup.async createWithCustomConfiguration):
(TestGroup.async createAndRefetchTestGroups):
* Websites/perf.webkit.org/public/v3/models/test-parameter.js: Added 'TestParameter' and 'TestParameterSet'
models which are JavaScript representations of 'test_paramters' and 'test_parameter_sets' tables.
Added 'CustomTestParameterSet' to be used in the UI.
(TestParameter):
(TestParameter.prototype.get disabled):
(TestParameter.prototype.get type):
(TestParameter.prototype.get description):
(TestParameter.prototype.get hasFile):
(TestParameter.prototype.get hasValue):
(TestParameter.findByName):
(TestParameterSet):
(TestParameterSet.prototype.updateSingleton):
(TestParameterSet.prototype._updateFromObject):
(TestParameterSet.prototype.get buildTypeParameters):
(TestParameterSet.prototype.get testTypeParameters):
(TestParameterSet.prototype.get parameters):
(TestParameterSet.prototype.asPayload):
(TestParameterSet.prototype.hasSameBuildParametersAs):
(TestParameterSet.prototype.equals):
(TestParameterSet.prototype.valueForParameter):
(TestParameterSet.prototype.fileForParameter):
(TestParameterSet.prototype.valueForParameterName):
(TestParameterSet.prototype.fileForParameterName):
(TestParameterSet.prototype._createPayloadForParameters):
(TestParameterSet.prototype._checkEquals):
(TestParameterSet.isValueEqual):
(TestParameterSet.equals):
(TestParameterSet.hasSameBuildParameters):
(CustomTestParameterSet):
(CustomTestParameterSet.prototype.get buildTypeParameters):
(CustomTestParameterSet.prototype.get testTypeParameters):
(CustomTestParameterSet.prototype.get parameters):
(CustomTestParameterSet.prototype.asPayload):
(CustomTestParameterSet.prototype.valueForParameter):
(CustomTestParameterSet.prototype.fileForParameter):
(CustomTestParameterSet.prototype.valueForParameterName):
(CustomTestParameterSet.prototype.fileForParameterName):
(CustomTestParameterSet.prototype.set):
(CustomTestParameterSet.prototype.get):
(CustomTestParameterSet.prototype.delete):
(CustomTestParameterSet.prototype.setByName):
(CustomTestParameterSet.prototype.getByName):
(CustomTestParameterSet.prototype._createPayloadForParameters):
* Websites/perf.webkit.org/public/v3/models/triggerable.js: Added 'testParamters' to 'TriggerableConfiguration' to reflect
corresponding database change.
(Triggerable):
(prototype.get testParameters):
(createForTriggerable):
* Websites/perf.webkit.org/public/v3/pages/analysis-task-page.js: Adopted APIs that create test groups now require test parameter sets to be specified.
(AnalysisTaskPage.prototype.async _retryCurrentTestGroup):
(AnalysisTaskPage.prototype.async _bisectCurrentTestGroup):
(AnalysisTaskPage.set const):
(AnalysisTaskPage.prototype.async _createTestGroupAfterVerifyingCommitSetList):
(AnalysisTaskPage.prototype.async _createCustomTestGroup):
* Websites/perf.webkit.org/public/v3/pages/create-analysis-task-page.js:
(CreateAnalysisTaskPage.prototype.async _createAnalysisTaskWithGroup):
* Websites/perf.webkit.org/server-tests/api-build-requests-tests.js:
(then):
(async await):
* Websites/perf.webkit.org/server-tests/api-manifest-tests.js:
(return.TestServer.remoteAPI.getJSON.string_appeared_here.then):
* Websites/perf.webkit.org/server-tests/api-test-groups.js:
(async const):
* Websites/perf.webkit.org/server-tests/api-update-triggerable-tests.js: Updated and added unit tests.
* Websites/perf.webkit.org/server-tests/privileged-api-add-build-requests-tests.js: Updated and added unit tests.
(async addTriggerableAndCreateTask):
(async await):
(webkit.id):
(const.commitSet.of.updatedGroups.0.requestedCommitSets):
* Websites/perf.webkit.org/server-tests/privileged-api-create-analysis-task-tests.js: Updated and added unit tests.
* Websites/perf.webkit.org/server-tests/privileged-api-create-test-group-tests.js: Updated and added unit tests.
(async await):
(webkit.id):
(async const):
* Websites/perf.webkit.org/server-tests/resources/mock-data.js: Added test parameters related mock data.
(MockData.addMockConfiguration):
(MockData.async await):
(MockData.set addMockDataWithBuildAndTestTypeTestParameterSets):
(MockData.set addMockDataWithTestParameterSets):
(MockData.mockTestSyncConfigWithTestParameters):
(MockData.addEmptyTriggerable): Deleted.
(MockData.addMockTestGroupWithGitWebKit): Deleted.
* Websites/perf.webkit.org/server-tests/tools-buildbot-triggerable-tests.js: Updated and added unit tests.
(async const):
* Websites/perf.webkit.org/server-tests/tools-sync-buildbot-integration-tests.js:
(configWithOneTesterTwoBuilders):
* Websites/perf.webkit.org/tools/js/buildbot-syncer.js:
(BuildbotSyncer):
(BuildbotSyncer.prototype.addTestConfiguration):
(BuildbotSyncer.prototype.addBuildConfiguration):
(BuildbotSyncer.prototype.matchesConfiguration):
(BuildbotSyncer.prototype._propertiesForBuildRequest):
(BuildbotSyncer._loadConfig):
(BuildbotSyncer._resolveBuildersWithPlatforms):
(BuildbotSyncer._validateAndMergeConfig):
* Websites/perf.webkit.org/tools/js/buildbot-triggerable.js: Updated and added unit tests.
(BuildbotTriggerable.prototype.updateTriggerable):
* Websites/perf.webkit.org/tools/js/database.js:
* Websites/perf.webkit.org/tools/js/v3-models.js:
* Websites/perf.webkit.org/unit-tests/build-request-tests.js: Updated and added unit tests.
(sampleBuildRequestData):
(oneTestGroup):
(threeTestGroups):
* Websites/perf.webkit.org/unit-tests/resources/mock-v3-models.js: Added mock test parameters.
* Websites/perf.webkit.org/unit-tests/test-groups-tests.js: Updated and added unit tests.
(sampleTestGroup):
(sampleTestGroupForRetry):
* Websites/perf.webkit.org/unit-tests/test-parameter-tests.js: Added unit tests.

Canonical link: https://commits.webkit.org/265477@main
